react/src/components/UserProfileSettingModal.tsx

```diff
@@ -0,0 +1,304 @@
+import React from "react";
+import { Modal, Input, Form, Select, SelectProps, message, Switch } from "antd";
+import { useTranslation } from "react-i18next";
+import { useWebComponentInfo } from "./DefaultProviders";
+import { passwordPattern } from "./ResetPasswordRequired";
+import { useSuspendedBackendaiClient } from "../hooks";
+import { useTanQuery, useTanMutation } from "../hooks/reactQueryAlias";
+import _ from "lodash";
+
+const UserProfileSettingModal: React.FC = () => {
+  const { t } = useTranslation();
+
+  const [form] = Form.useForm();
+
+  const [messageApi, contextHolder] = message.useMessage();
+  const _showSuccessMessage = (successMessage: string) => {
+    messageApi.open({
+      type: "success",
+      content: successMessage,
+    });
+  };
+  const _showErrorMessage = (errorMessage: string) => {
+    messageApi.open({
+      type: "error",
+      content: errorMessage,
+    });
+  };
+
+  const baiClient = useSuspendedBackendaiClient();
+  let email = baiClient.email;
+  let full_name = baiClient.full_name;
+  let loggedAccount = baiClient._config.accessKey;
```

Comment:
You don't need to defined every variable to use this component. You can use it directly where it neeeds. It's better to increae readability.